scipy/ndimage/tests/test_morphology.py

```diff
@@ -147,45 +169,49 @@ def test_distance_transform_bf03(self, dtype):
         assert_array_almost_equal(ft, expected)
 
     @pytest.mark.parametrize('dtype', types)
-    def test_distance_transform_bf04(self, dtype):
-        data = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 1, 1, 1, 0, 0, 0],
-                         [0, 0, 1, 1, 1, 1, 1, 0, 0],
-                         [0, 0, 1, 1, 1, 1, 1, 0, 0],
-                         [0, 0, 1, 1, 1, 1, 1, 0, 0],
-                         [0, 0, 0, 1, 1, 1, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype)
+    def test_distance_transform_bf04(self, dtype, xp):
+        dtype = getattr(xp, dtype)
+        if is_cupy(xp):
+            pytest.xfail("CuPy does not have distance_transform_bf.")
+
+        data = xp.asarray([[0, 0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 1, 1, 1, 0, 0, 0],
+                           [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                           [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                           [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                           [0, 0, 0, 1, 1, 1, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=dtype)
         tdt, tft = ndimage.distance_transform_bf(data, return_indices=1)
         dts = []
         fts = []
-        dt = np.zeros(data.shape, dtype=np.float64)
+        dt = xp.zeros(data.shape, dtype=xp.float64)
         ndimage.distance_transform_bf(data, distances=dt)
         dts.append(dt)
         ft = ndimage.distance_transform_bf(
             data, return_distances=False, return_indices=1)
         fts.append(ft)
-        ft = np.indices(data.shape, dtype=np.int32)
+        ft = np.indices(data.shape, dtype=xp.int32)
```

Comment:
This `cupy`-specific change makes me a bit uncomfortable, but sure, we can leave it to the follow-up.